resources.py

```diff
@@ -910,6 +910,53 @@ def enable_dhcp_in_networkd(self, interface):
         # Wait a few seconds for the DHCP to be applied
         time.sleep(5)
 
+    def prepare_nested_virtualization(self):
+        """ Prepare all packages to allow kvm virtualization. """
+
+        # Install the required package
+        self.run('sudo apt update')
+        self.run('sudo apt install -y libvirt-clients')
+
+        return self.run('sudo virt-host-validate').stdout
+
+    def setup_nested_virtualization(self, vm_os, vm_iso_url):
+        """ Create a KVM virtual machine. """
+
+        # Install the required package
+        self.prepare_nested_virtualization()
+        self.run(oneliner(f"""
+            sudo apt install -y
+                qemu-kvm
+                libvirt-daemon-system
+                bridge-utils
+                virt-manager
+        """))
+
+        # Make sure qemu tests pass and rc == 0
+        assert self.output_of('sudo virt-host-validate --help')
+
+        self.run(f'wget {vm_iso_url}')
+        self.run('sudo virsh net-start default')
+        self.run('sudo virsh net-autostart default')
+        self.run(f'qemu-img create -f qcow2 {vm_os}.qcow2 8G')
+        self.run(oneliner(f"""
```

Comment:
If we use our own custom alpine image, we could start that, then get the console output and check that it boots.